engine/classes/Elgg/Router.php

```diff
@@ -70,42 +72,51 @@ public function route(\Elgg\Http\Request $request) {
 			$this->timer->begin(['build page']);
 		}
 
+		ob_start();
+
 		$result = $this->hooks->trigger('route', $identifier, $old, $old);
 		if ($result === false) {
-			return true;
-		}
+			$output = ob_get_clean();
+			$response = elgg_ok($output);
+		} else {
+			if ($result !== $old) {
+				_elgg_services()->logger->warn('Use the route:rewrite hook to modify routes.');
+				$query = parse_url(current_page_url(), PHP_URL_QUERY);
+				$segments = (array) elgg_extract('segments', $result, []);
+				$identifier = elgg_extract('identifier', $result);
+				array_unshift($segments);
+				_elgg_services()->responseFactory->redirect(implode('/', $segments) . '?' . $query);
```

Comment:
This is not BC. Need to rewrite current route in BC way, so that `ResponseFactory` can parse response type.